File: APP03022-mrch.backend.somx.fiscal-api/src/main/java/com/sodimac/fiscal/api/model/entity/VersionCatalogEntity.java
package com.sodimac.fiscal.api.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.math.BigDecimal;

@Entity
@Table(name = "version_catalog")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class VersionCatalogEntity extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "version_id")
    private Long versionId;

    @Column(name = "name", length = 50, nullable = false)
    private String name;

    @Column(name = "description", length = 254)
    private String description;

    @Column(name = "version", precision = 7, scale = 3, nullable = false)
    private BigDecimal version;

    @Column(name = "document_type", length = 1, nullable = false)
    private String documentType;

    @Column(name = "pac_id", nullable = false)
    private Integer pacId;

    @Column(name = "valid_from")
    private LocalDateTime validFrom;

    @Column(name = "valid_to")
    private LocalDateTime validTo;

    @Column(name = "structure_url", length = 254)
    private String structureUrl;

    @Column(name = "status")
    private Integer status = 1;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pac_id", insertable = false, updatable = false)
    private PacCatalogEntity pacCatalog;
}

File: APP03022-mrch.backend.somx.finanzas-api/src/entities/VersionCatalog.entity.ts
import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    CreateDateColumn,
    UpdateDateColumn,
    ManyToOne,
    JoinColumn
} from 'typeorm';
import { PacCatalog } from './PacCatalog.entity.js';

@Entity('version_catalog')
export class VersionCatalog {
    @PrimaryGeneratedColumn({ name: 'version_id' })
    versionId!: number;

    @Column({ type: 'varchar', length: 50, nullable: false })
    name!: string;

    @Column({ type: 'varchar', length: 254, nullable: true })
    description?: string;

    @Column({ type: 'numeric', precision: 7, scale: 3, nullable: false })
    version!: number;

    @Column({ name: 'document_type', type: 'char', length: 1, nullable: false })
    documentType!: string;

    @Column({ name: 'pac_id', type: 'integer', nullable: false })
    pacId!: number;

    @Column({ name: 'valid_from', type: 'timestamp', nullable: true })
    validFrom?: Date;

    @Column({ name: 'valid_to', type: 'timestamp', nullable: true })
    validTo?: Date;

    @Column({ name: 'structure_url', type: 'varchar', length: 254, nullable: true })
    structureUrl?: string;

    @Column({ type: 'integer', default: 1 })
    status!: number;

    @Column({ name: 'created_by', type: 'bigint', nullable: true })
    createdBy?: number;

    @CreateDateColumn({ name: 'created_at', type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
    createdAt!: Date;

    @Column({ name: 'updated_by', type: 'bigint', nullable: true })
    updatedBy?: number;

    @UpdateDateColumn({ name: 'updated_at', type: 'timestamp', nullable: true })
    updatedAt?: Date;

    @ManyToOne(
        () => PacCatalog,
        (pac) => pac.versions
    )
    @JoinColumn({ name: 'pac_id' })
    pac!: any;

}
